File: trees/101.js
class Solution{
    constructor(val = 0, left = null, right = null){
        this.val = val
        this.left = left
        this.right = right
    }
}
class Solution{
    check(l, r){
        if(!l && !r){
            return true
        }
        if(!l || !r){
            return false
        }
        return (l.val === r.val) && this.check(l.left, r.right) && this.check(l.right === r.left)
    }
    isSymmetric(root){
        if(!root) return true;
        return this.check(root.left, root.right)
    }
}

File: trees/101.py
# Tc = O(n) [need to visit every node in the symmetry]
# Sc = O(h) [max depth of recursion stack, in the case of balanced binary tree the height will be O(log.n)]

class Solution:
    def check(self, l, r):
        if not l and not r:
            return True
        if not l or not r:
            return False
        if l.val == r.val and l.left == r.right and l.right == r.left:
            return True
        return False
    def isSymmetric(self,root):
        if not root:
            return True
        return self.check(root.left, root.right)